homeassistant/components/luxtronik/sensor.py

```diff
@@ -0,0 +1,192 @@
+        return False
+
+    sensors = config.get(CONF_SENSORS)
+
+    entities = []
+    for sensor_cfg in sensors:
+        sensor = luxtronik.get_sensor(sensor_cfg[CONF_GROUP], sensor_cfg[CONF_ID])
+        if sensor:
+            entities.append(LuxtronikSensor(luxtronik, sensor, sensor_cfg))
+        else:
+            _LOGGER.warning(
+                "Invalid Luxtronik ID %s in group %s",
+                sensor_cfg[CONF_ID],
+                sensor_cfg[CONF_GROUP],
+            )
+
+    add_entities(entities, True)
+
+
+class LuxtronikSensor(Entity):
+    """Representation of a Luxtronik sensor."""
+
+    def __init__(self, luxtronik, sensor, sensor_cfg):
+        """Initialize a new Luxtronik sensor."""
+        self._luxtronik = luxtronik
+        self._sensor = sensor
+        self._name = sensor_cfg[CONF_FRIENDLY_NAME]
+        self._icon = sensor_cfg[CONF_ICON]
+
+    @property
+    def entity_id(self):
+        """Return the entity_id of the sensor."""
+        if not self._name:
+            return ENTITY_ID_FORMAT.format(slugify(self._sensor.name))
+        return ENTITY_ID_FORMAT.format(slugify(self._name))
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        if not self._name:
+            return self._sensor.name
+        return self._name
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        if not self._icon:
+            icons = {
+                CONF_CELSIUS: "mdi:thermometer",
+                CONF_SECONDS: "mdi:timer-sand",
+                CONF_PULSES: "mdi:pulse",
+                CONF_IPADDRESS: "mdi:ip-network-outline",
+                CONF_TIMESTAMP: "mdi:calendar-range",
+                CONF_ERRORCODE: "mdi:alert-circle-outline",
+                CONF_KELVIN: "mdi:thermometer",
+                CONF_BAR: "mdi:arrow-collapse-all",
+                CONF_PERCENT: "mdi:percent",
+                CONF_RPM: "mdi:rotate-right",
+                CONF_ENERGY: "mdi:flash-circle",
+                CONF_VOLTAGE: "mdi:flash-outline",
+                CONF_HOURS: "mdi:clock-outline",
+                CONF_FLOW: "mdi:chart-bell-curve",
+                CONF_LEVEL: "mdi:format-list-numbered",
+                CONF_COUNT: "mdi:counter",
+                CONF_VERSION: "mdi:information-outline",
+            }
+            return icons.get(self._sensor.measurement_type, None)
+        return self._icon
+
+    @property
+    def state(self):
+        """Return the sensor state."""
+        return self._sensor.value
+
+    @property
+    def device_class(self):
+        """Return the class of this sensor."""
+        device_classes = {
+            CONF_CELSIUS: DEVICE_CLASS_PRESSURE,
+            CONF_KELVIN: DEVICE_CLASS_PRESSURE,
+            CONF_BAR: DEVICE_CLASS_PRESSURE,
+            CONF_SECONDS: DEVICE_CLASS_TIMESTAMP,
+            CONF_HOURS: DEVICE_CLASS_TIMESTAMP,
+            CONF_TIMESTAMP: DEVICE_CLASS_TIMESTAMP,
+        }
```

Comment:
I would recommend to move this up to the top as well and replace the custom constants with strings, e.g.
https://github.com/home-assistant/home-assistant/blob/d851cb6f9e116dddb8ebeed4dd3910eb811993bd/homeassistant/components/alpha_vantage/sensor.py#L28-L36

Reply:
Should I maybe move ICONS to const.py as well an import it?

Reply:
Yeah that is a good idea, move ICONS and device_classes to const.py